### ISSUE TITLE
PM2 Low memory environment options

### ISSUE THREAD

**Issue by Unitech** [Unitech/pm2#2501]
Refering to http://fiznool.com/blog/2016/10/01/running-a-node-dot-js-app-in-a-low-memory-environment/

We could offer an option that could make PM2 well fitted for low-memory environment. We would mainly auto-tune `max_old_space_size` depending on the memory available.
By default PM2 use as low as 30mb memory, we could enhance this.

**Comment by Unitech** [comment]
configurable via

```
$ PM2_OPTIMIZE_MEMORY=true pm2 update
```

**Comment by Unitech** [comment]
Published on PM2 2.6.0:

```bash
$ npm install pm2@latest -g
$ pm2 update
```

**Comment by jiangbodev** [comment]
how to use this `PM2_OPTIMIZE_MEMORY` ?
